core/src/nats.rs

```diff
@@ -450,24 +450,29 @@ impl TypedNats {
         T: TypedMessage,
     {
         let bytes = Bytes::from(serde_json::to_vec(value)?);
-        let result1 = self
-            .nc
-            .request(value.subject(), bytes.clone());
-        
-        let result = if let Some(tmp_alt_subject) = value.tmp_alt_subject() {
-            let result2 = self.nc
-                .request(tmp_alt_subject, bytes);
-
-            select! {
-                result = result1 => result.to_anyhow()?,
-                result = result2 => result.to_anyhow()?,
-            }
-        } else {
-            result1.await.to_anyhow()?
+
+        let mut futs = FuturesUnordered::new();
```

Comment:
Pushing a future to `futs` here will immediately call `.await` on it (i.e. "fire" it off)? Or does `futs.next()` fire them all off? Or something else?

Reply:
Awaiting futs.next() will poll both promises, and then will return the result of each in the order it resolves.